app/services/search_indexer.rb

```diff
@@ -120,11 +120,11 @@ def self.update_posts_index(post_id:, topic_title:, category_name:, topic_tags:,
       a_weight: topic_title,
       b_weight: category_name,
       c_weight: topic_tags,
-      # Length of a tsvector must be less than 1_048_576 bytes.
-      # The difference between the max ouptut limit and imposed input limit
-      # accounts for the fact that sometimes the output tsvector may be
-      # slighlty longer than the input.
-      d_weight: scrub_html_for_search(cooked)[0..1_000_000]
+      # The tsvector resulted from parsing a string can be with 10% longer
```

Comment:
Where did you see that a ts_vector can be 10% longer?

Reply:
I have updated the figure. I have determined empirically that a more appropriate limit is 660k. A string of 660k characters composed of unique words (`a b c ... aa ab ac ...`) can generate a tsvector 1MB long.